tools/cli/factory.go

```diff
@@ -70,35 +70,31 @@ type ClientFactory interface {
 	ServerFrontendClient(c *cli.Context) frontend.Client
 	ServerAdminClient(c *cli.Context) admin.Client
 
+	// ServerFrontendClientForMigration frontend client of the migration destination
+	ServerFrontendClientForMigration(c *cli.Context) frontend.Client
+	// ServerAdminClientForMigration admin client of the migration destination
+	ServerAdminClientForMigration(c *cli.Context) admin.Client
+
 	ElasticSearchClient(c *cli.Context) *elastic.Client
 
 	ServerConfig(c *cli.Context) (*config.Config, error)
 }
 
 type clientFactory struct {
-	addressFlagFunc func(c *cli.Context) string
-	hostPort        string
-	dispatcher      *yarpc.Dispatcher
-	logger          *zap.Logger
+	dispatcher          *yarpc.Dispatcher
+	dispatcherMigration *yarpc.Dispatcher
+	logger              *zap.Logger
 }
 
-// DEPRECATED don't use, only reserved for backward compatibility purposes
 // NewClientFactory creates a new ClientFactory
 func NewClientFactory() ClientFactory {
```

Comment:
The method was marked as deprecated, is there a way to not use it?

Reply:
it was deprecated in the previous CLI pr but then I realize it should not be deprecated